app/controllers/thread_controller.php

```diff
@@ -1,35 +1,180 @@
+        else redirect(url('user/login_notice'));
     }   
 
-    public function view()                
+    public function general_category()
     {
-        $thread = Thread::get(Param::get('thread_id'));
-        $comments = $thread->getComments();
+        $threads = Thread::getThreadsByCategory("General");
         $this->set(get_defined_vars());
     }
 
+    public function technology_category()
+    {
+
+    }
+
+    public function view()                
+    {
+        if(isset($_SESSION["username"])) 
+        {
+            $comments_per_page = 10;
+            $pageOne = 1;
+            $thread = Thread::get(Param::get('thread_id'));
+            $thread_id = Param::get('thread_id');
+            $owner = Thread::getOwnerId($thread_id);
+            $ownerId = $owner["creator_id"];
+            $ownerUsername = User::getUsername($ownerId);
+            $comments = $thread->getComments();        
+            $page = Param::get("page",$pageOne); 
+            $pagination = new SimplePagination($page, $comments_per_page);
+            $comments = Comment::getAll($thread_id, $pagination->start_index - 1, $pagination->count + 1);
+            $pagination->checkLastPage($comments);
+            $total = Comment::countAll($thread_id);
+            $pages = ceil($total / $comments_per_page);
+            $threadContent = Thread::getBodyContents((int)$thread_id);
+            $this->set(get_defined_vars());
+        }
+        else redirect(url('user/login_notice'));
+    }
+
+    public function edit()
+    {
+        if(isset($_SESSION["username"]))
+        {
+            $thread = Thread::get(Param::get('thread_id'));
+            $page = Param::get('page_next', 'edit');
+            $_SESSION["title"] = Param::get('title');
+            $_SESSION["threadBody"] = Param::get('thread_body');
+            switch ($page) {
+            case self::EDIT:                    
+                break;
+            case self::EDIT_END:  
+                $title                      = Param::get('title');
+                $threadBody                 = Param::get('thread_body');
+                $thread->new_title          = $title;
+                $thread->new_thread_body    = $threadBody;
+                try {            
+                        $thread->update(Param::get('thread_id'));
+                } catch (ValidationException $e) {                    
+                       $page = 'edit';
+                    }                           
+                break;
+            default:                    
+                throw new NotFoundException("{$page} is not found");        
+                break;
+            }
+            $this->set(get_defined_vars());
+            $this->render($page);
+        }
+        else redirect(url('user/login_notice'));
+    }
+
+    public function create()
+    { 
+        if(isset($_SESSION["username"])) 
+        {
+            $thread = new Thread;
+            $comment = new Comment;
```

Comment:
Same as my comment here https://github.com/lawrence-agustin/Interconnect/pull/2#discussion_r30960756, 

This should be
```
 $thread = new Thread();
 $comment = new Comment();
```